cluster-autoscaler/cloudprovider/konvoy/konvoy.go

```diff
@@ -91,11 +78,12 @@ func (konvoy *KonvoyCloudProvider) GetAvailableGPUTypes() map[string]struct{} {
 
 // NodeGroups returns all node groups configured for this cloud provider.
 func (konvoy *KonvoyCloudProvider) NodeGroups() []cloudprovider.NodeGroup {
-	nodeGroups := make([]cloudprovider.NodeGroup, len(konvoy.konvoyManager.GetNodeGroups()))
-	for i, ng := range konvoy.konvoyManager.GetNodeGroups() {
-		nodeGroups[i] = ng
+	ngs := konvoy.konvoyManager.GetNodeGroups()
+	out := make([]cloudprovider.NodeGroup, len(ngs))
+	for i, ng := range ngs {
+		out[i] = ng
 	}
```

Comment:
https://github.com/golang/go/wiki/SliceTricks#copy
```suggestion
	copy(out, ngs)
```